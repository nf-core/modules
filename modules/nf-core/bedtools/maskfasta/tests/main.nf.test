
nextflow_process {

    name "Test Process BEDTOOLS_MASKFASTA"
    script "../main.nf"
    process "BEDTOOLS_MASKFASTA"

    tag "modules"
    tag "modules_nfcore"
    tag "bedtools"
    tag "bedtools/maskfasta"

    test("test-bedtools-maskfasta") {

        when {
            process {
                """
                input[0] = [ [ id:'test'],
				    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)
				]
				input[1] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
