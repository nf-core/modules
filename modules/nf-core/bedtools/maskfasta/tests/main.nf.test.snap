{
    "test-bedtools-maskfasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,f4f6749698f11074228d2c79338e3b9c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,441689b51fc99b551e274857bb36445a"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,f4f6749698f11074228d2c79338e3b9c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,441689b51fc99b551e274857bb36445a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-26T13:57:44.016613"
    }
}